types/format_test.go

```diff
@@ -118,6 +119,12 @@ func (s *testTimeSuite) TestStrToDate(c *C) {
 		{`70/10/22`, `%Y/%m/%d`, types.FromDate(1970, 10, 22, 0, 0, 0, 0)},
 		{`18/10/22`, `%Y/%m/%d`, types.FromDate(2018, 10, 22, 0, 0, 0, 0)},
 		{`100/10/22`, `%Y/%m/%d`, types.FromDate(100, 10, 22, 0, 0, 0, 0)},
+		{`200442 Monday`, `%X%V %W`, types.FromDate(2004, 10, 18, 0, 0, 0, 0)},
+		{`2004420`, `%X%V%w`, types.FromDate(2004, 10, 17, 0, 0, 0, 0)},
+		{`2004423`, `%X%V%w`, types.FromDate(2004, 10, 20, 0, 0, 0, 0)},
+		{`200442 Sunday`, `%x%v%W`, types.FromDate(2004, 10, 17, 0, 0, 0, 0)},
```

Comment:
``` sql
mysql> select str_to_date("200442 S","%x%v%W");
+----------------------------------+
| str_to_date("200442 S","%x%v%W") |
+----------------------------------+
| NULL                             |
+----------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> select str_to_date("200442 Sun","%x%v%W");
+------------------------------------+
| str_to_date("200442 Sun","%x%v%W") |
+------------------------------------+
| 2004-10-17                         |
+------------------------------------+
1 row in set (0.00 sec)

mysql> select str_to_date("200442 sun","%x%v%W");
+------------------------------------+
| str_to_date("200442 sun","%x%v%W") |
+------------------------------------+
| 2004-10-17                         |
+------------------------------------+
1 row in set (0.00 sec)
```
I think we can add some tests for these.

Reply:
Done